### PR TITLE
[CONFIGURE] Minor configuration fixes for Windows

### PR DESCRIPTION
**Pull request by luizfernandonb**
## Purpose
Fix some things in the configuration file that I believe are wrong.

## Proposed changes
- Apart from making it a little more user-friendly to read, there is a change in checking when the configuration is for `VSSolution`, where the file was not correctly checking if the solution was in that folder.

- And I made the `-S` flag of source explicit when the file calls CMake to avoid possible configuration errors on the part of CMake.

### CODE REVIEW

**Comment by ColinFinck** [comment]
Can you give an example of what is fixed by specifying `-S` explicitly?

**Comment by luizfernandonb** [comment]
> Can you give an example of what is fixed by specifying `-S` explicitly?

Not at the moment, but when I was analyzing this file, I realized that the amount of arguments could cause some confusion, both for those who read the file and when for CMake, the -S does not influence anything, it only specifies the source directory, For me, it's even better to understand.

File: configure.cmd [hunk 1]
```diff
@@ -85,12 +92,14 @@ REM Parse command line parameters
...
-            echo. && echo Error: Creation of VS Solution files is not supported in a MinGW environment.
+            echo. && echo Error: Creation of Visual Studio solution files is not supported in a MinGW environment.
             echo Please run this command in a [Developer] Command Prompt for Visual Studio.
+
             goto quit
         ) else if /I "%1" NEQ "" (
             echo.%1| find /I "-D" >nul 2>&1
-            if not errorlevel 1 (
+            
+			if not errorlevel 1 (
```
Comment (learn-more) [inline configure.cmd:102]: You are changing a lot of whitespace around for no good reason?

File: configure.cmd [hunk 2]
```diff
@@ -111,12 +120,15 @@ REM Parse command line parameters
             set USE_CLANG_CL=1
         ) else if /I "%1" == "VSSolution" (
             set VS_SOLUTION=1
-            REM explicitly set VS version for project generator
+            
+			REM explicitly set VS version for project generator
```
Comment (HBelusca) [inline configure.cmd:124]: Don't use tabs.

Comment (luizfernandonb) [reply configure.cmd:124]: Nothing has changed, what's different about this line? It's just a space down more.

File: configure.cmd (hunk 1 above)
Comment (binarymaster) [reply configure.cmd:94]: Tabs detected on this line.

File: configure.cmd (hunk 2 above)
Comment (binarymaster) [reply configure.cmd:115]: And there still.

File: configure.cmd [hunk 3]
```diff
@@ -159,56 +160,59 @@ if "%VS_SOLUTION%" == "1" (
...
-) else if %USE_CLANG_CL% == 1 (
-    cmake -G %CMAKE_GENERATOR% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% -DUSE_CLANG_CL:BOOL=1 %* "%REACTOS_SOURCE_DIR%"
+    cmake -G %CMAKE_GENERATOR% -DENABLE_CCACHE:BOOL=0 -DCMAKE_TOOLCHAIN_FILE:FILEPATH=%MINGW_TOOCHAIN_FILE% -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% %* -S "%REACTOS_SOURCE_DIR%"
 ) else (
-    cmake -G %CMAKE_GENERATOR% %CMAKE_ARCH% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% %* "%REACTOS_SOURCE_DIR%"
+    if "%USE_CLANG_CL%" == "1" (
+		cmake -G %CMAKE_GENERATOR% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% -DUSE_CLANG_CL:BOOL=1 %* -S "%REACTOS_SOURCE_DIR%"
+    ) else (
+		cmake -G %CMAKE_GENERATOR% %CMAKE_ARCH% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% %* -S "%REACTOS_SOURCE_DIR%"
+    )
```
Comment (binarymaster) [inline configure.cmd:201]: Tabs on lines 198 and 200.

**Review by binarymaster — COMMENTED** [review]
I'd recommend using Notepad++ with "Show All Characters" option, so you can easily spot unwanted tabs.

File: configure.cmd [hunk 4]
```diff
@@ -159,56 +160,59 @@ if "%VS_SOLUTION%" == "1" (
...
+    if exist "REACTOS.sln" (
+        echo.
+		echo Error: This directory has already been configured for Visual Studio, delete the contents of this folder, then try again.
+
+        goto quit
+    )
+) else (
+    if exist "build.ninja" (
+        echo.
+		echo Error: This directory has already been configured for ninja, delete the contents of this folder, then try again.
```
Comment (binarymaster) [inline configure.cmd:182]: Tabs on 175 and 182.

**Review by binarymaster — COMMENTED** [review]
No more tabs. My job here is done.

File: configure.cmd [hunk 5]
```diff
@@ -159,56 +160,59 @@ if "%VS_SOLUTION%" == "1" (
...
+    if not exist "%REACTOS_OUTPUT_PATH%" (
+        echo Creating directories...
+
         mkdir %REACTOS_OUTPUT_PATH%
     )
     cd %REACTOS_OUTPUT_PATH%
 )
 
 if "%VS_SOLUTION%" == "1" (
+    if exist "REACTOS.sln" (
```
Comment (Extravert-ir) [inline configure.cmd:173]: You have reverted the logic here, please read carefully and return back

File: configure.cmd [hunk 6]
```diff
@@ -159,56 +160,59 @@ if "%VS_SOLUTION%" == "1" (
...
 
-
 if "%BUILD_ENVIRONMENT%" == "MinGW" (
-    cmake -G %CMAKE_GENERATOR% -DENABLE_CCACHE:BOOL=0 -DCMAKE_TOOLCHAIN_FILE:FILEPATH=%MINGW_TOOCHAIN_FILE% -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% %* "%REACTOS_SOURCE_DIR%"
-) else if %USE_CLANG_CL% == 1 (
-    cmake -G %CMAKE_GENERATOR% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% -DUSE_CLANG_CL:BOOL=1 %* "%REACTOS_SOURCE_DIR%"
+    cmake -G %CMAKE_GENERATOR% -DENABLE_CCACHE:BOOL=0 -DCMAKE_TOOLCHAIN_FILE:FILEPATH=%MINGW_TOOCHAIN_FILE% -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% %* -S "%REACTOS_SOURCE_DIR%"
 ) else (
-    cmake -G %CMAKE_GENERATOR% %CMAKE_ARCH% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% %* "%REACTOS_SOURCE_DIR%"
+    if "%USE_CLANG_CL%" == "1" (
```
Comment (Extravert-ir) [inline configure.cmd:197]: Please, leave `) else if` statement as it was

**Review by Extravert-ir — CHANGES_REQUESTED** [review]

**Pull request closed without merging**